backend/models/databases/supabase/user_usage.py

```diff
@@ -25,31 +26,69 @@ def create_user_daily_usage(self, user_id: UUID, user_email: str, date: datetime
             .execute()
         )
 
+    def check_if_is_premium_user(self, user_id: UUID):
+        """
+        Check if the user is a premium user
+        """
+        user_email_customer = (
+            self.db.from_("users")
+            .select("*")
+            .filter("id", "eq", str(user_id))
+            .execute()
+        ).data
+
+        if len(user_email_customer) == 0:
+            return False
+
+        matching_customers = (
+            self.db.table("customers")
+            .select("email")
+            .filter("email", "eq", user_email_customer[0]["email"])
+            .execute()
+        ).data
+
+        return len(matching_customers) > 0
+
     def get_user_settings(self, user_id):
         """
         Fetch the user settings from the database
         """
-        response = (
+
+        user_settings_response = (
             self.db.from_("user_settings")
             .select("*")
             .filter("user_id", "eq", str(user_id))
             .execute()
         ).data
 
-        if len(response) == 0:
+        if len(user_settings_response) == 0:
             # Create the user settings
-            result = (
+            user_settings_response = (
                 self.db.table("user_settings")
                 .insert({"user_id": str(user_id)})
                 .execute()
+            ).data
+
+        if len(user_settings_response) == 0:
+            raise ValueError("User settings could not be created")
+
+        user_settings = user_settings_response[0]
+        user_settings["is_premium"] = False
+        is_premium_user = self.check_if_is_premium_user(user_id)
+
+        if is_premium_user:
+            user_settings["is_premium"] = True
+            user_settings["max_brains"] = int(
+                os.environ.get("PREMIUM_MAX_BRAIN_NUMBER", 30)
```

Comment:
The value is already in the database. Don't use the env variable. Use the value from the  database instead. 

It allows for example to set custom values to not paying customers